### ISSUE TITLE
Browser loses connection to kernel

### ISSUE THREAD

**Issue by aamster** [jupyterlab/jupyterlab#7556]
## Description

Browser periodically loses connection to kernel, sometimes resulting in loss of work.
Get the "Aw snap" page on Chrome: https://support.google.com/chrome/answer/95669?visit_id=637100456501418614-4256533815&p=e_awsnap&rd=1. It randomly occurs. Not sure how to reproduce it. 

## Reproduce

Not sure how to reproduce it.

## Context

- Operating System and version: Linux Centos 7
- Browser and version: chrome latest
- JupyterLab version: 1.0

<details><summary>Command Line Output</summary>
<pre>
```
[I 13:51:40.270 LabApp] Starting buffering for a33aac91-9a74-42ed-9278-5fefea681e9d:30a65336-6483-41c5-bca4-79442dd3cc69
[D 13:51:40.270 LabApp] Clearing buffer for a33aac91-9a74-42ed-9278-5fefea681e9d
[D 13:51:40.271 LabApp] Websocket closed 87ed3fa0-c555-4cf0-910f-57952267157a:52bdc595-429a-40af-b944-3b0cacc734dc
[I 13:51:40.271 LabApp] Starting buffering for 87ed3fa0-c555-4cf0-910f-57952267157a:52bdc595-429a-40af-b944-3b0cacc734dc
[D 13:51:40.272 LabApp] Clearing buffer for 87ed3fa0-c555-4cf0-910f-57952267157a
[D 13:54:16.566 LabApp] 200 GET /static/lab/vendors~main.07f31b9a0568bfe2120b.js (127.0.0.1) 1857.96ms

```
</pre>
</details>

**Comment by jasongrout** [comment]
It sounds like this might be a network issue? Do you have any other network issues when this happens?

**Comment by aamster** [comment]
It might be a network issue, but I don't notice any other loss of connectivity or any other network issues.

**Comment by jasongrout** [comment]
Did the server shut down for some reason? Did the browser have any errors?

I don't remember hearing about anything like this before. If you can get a way to reproduce this, that would be very helpful.

**Comment by jasongrout** [comment]
(Setting as needs info pending a way to reproduce this)

**Comment by aamster** [comment]
The server did not shut down. I pasted the only output in debug mode in the description. I will check next time it happens if there are any console errors.

**Comment by aamster** [comment]
Unfortunately there is no browser console output because devtools is disconnected when it crashes. Sorry, not sure what other info to provide to reproduce it.

**Comment by jasongrout** [comment]
Closing since there doesn't seem to be a way to reproduce this or move forward with it. Please comment below if you'd like to continue the discussion, or if you've continued to see this issue and can give any more clues about it.

**Comment by konradsitarz** [comment]
I had this issue with the `jupyterlab v1.2.3` on Firefox browser after ~20min of running cell, on Chrome after ~40minutes.

I deployed it on AKS with helm chart https://jupyterhub.github.io/helm-chart/jupyterhub-0.9.0-alpha.1.073.53374ee.tgz.
I used nginx ingress controller on k8s cluster. 

I suppose it might be connected with the nginx closing web socket too early but I don't have a clue how to debug it.
It also might be related to JupyterLab maybe it should reconnect to the kernel?
